crates/arroyo-connectors/src/filesystem/sink/parquet.rs

```diff
@@ -61,17 +70,25 @@ fn writer_properties_from_table(table: &FileSystemTable) -> WriterProperties {
 struct SharedBuffer {
     /// The inner buffer for reading and writing
     ///
-    /// The lock is used to obtain internal mutability, so no worry about the
+    /// The lock is used to obtain shared internal mutability, so no worry about the
     /// lock contention.
-    buffer: Arc<futures::lock::Mutex<Vec<u8>>>,
+    buffer: Arc<Mutex<bytes::buf::Writer<BytesMut>>>,
```

Comment:
The previous mutex was async-safe but the mutex is now `std::sync::Mutex`. I'm assuming this change is OK since the lock isn't being held across await points?

Reply:
Yes, there is no reason now for it to be a futures::lock::Mutex; I'd guess the structure of the code was different when this was added